custom_md_loader/classes/MetadataApexApiLoader.cls

```diff
@@ -0,0 +1,230 @@
+    	buildResponse();
+    }
+    
+    public override void migrateCustomMapping(String csName, String cmtName, String mapping) {
+    	super.migrateCustomMapping(csName, cmtName, mapping);
+    	buildResponse();
+    }
+    
+    private void buildResponse() {
+    	if(response.IsSuccess()) {
+			List<MetadataResponse.Message> messages = new List<MetadataResponse.Message>();
+			messages.add(new MetadataResponse.Message(100, 'Migration In Progress... Job Id: ' + getMdDeployStatus().getJobId()));
+			response.setIsSuccess(true);
+		    response.setMessages(messages);  
+		}
+    }
+    
+    public override void migrate(MetadataMappingInfo mappingInfo) {
+    	
+    	System.debug('MetadataApexApiLoader.migrate -->');
+    	try{
+    		Map<String, Schema.DescribeFieldResult> descFieldResultMap = mappingInfo.getSrcFieldResultMap();
+			String typeDevName = mappingInfo.getCustomMetadadataTypeName()
+						.subString(0, mappingInfo.getCustomMetadadataTypeName().indexOf(AppConstants.MDT_SUFFIX));
+	        List<Metadata.CustomMetadata> records = new List<Metadata.CustomMetadata>();            
+			for(sObject csRecord : mappingInfo.getRecordList()) {
+			
+	            Metadata.CustomMetadata customMetadataRecord = new Metadata.CustomMetadata();
+	            customMetadataRecord.values = new List<Metadata.CustomMetadataValue>();
+	            
+	            if(csRecord.get(AppConstants.CS_NAME_ATTRIBURE) != null) {
+	            	String strippedLabel = (String)csRecord.get(AppConstants.CS_NAME_ATTRIBURE);
+	            	String tempVal = strippedLabel.substring(0, 1);
+	                
+	                if(tempVal.isNumeric()) {
+	                	strippedLabel = 'X' + strippedLabel;
+	                }
+	            	strippedLabel = strippedLabel.replaceAll('\\W+', '_').replaceAll('__+', '_').replaceAll('\\A[^a-zA-Z]+', '').replaceAll('_$', '');
+	                System.debug('strippedLabel ->' + strippedLabel);
+	                
+	                // default fullName to type_dev_name.label
+					customMetadataRecord.fullName = typeDevName + '.'+ strippedLabel;
+					customMetadataRecord.label = (String)csRecord.get(AppConstants.CS_NAME_ATTRIBURE);
+	            }
+	            for(String fieldName : mappingInfo.getCSToMDT_fieldMapping().keySet()) {
+		            Schema.DescribeFieldResult descCSFieldResult = descFieldResultMap.get(fieldName.toLowerCase());
+	            
+	            	if(mappingInfo.getCSToMDT_fieldMapping().get(fieldName).endsWith('__c')){
+	        			Metadata.CustomMetadataValue cmv = new Metadata.CustomMetadataValue();
+	                	cmv.field = mappingInfo.getCSToMDT_fieldMapping().get(fieldName);
+	                	if(descCSFieldResult.getType().name() == 'DATETIME') {
+	                	
+	                		if(csRecord.get(fieldName) != null) {
+								Datetime dt = DateTime.valueOf(csRecord.get(fieldName));
+								String formattedDateTime = dt.format('yyyy-MM-dd\'T\'HH:mm:ss.SSSZ'); 

```

Comment:
This might not be the user's desired date format. Is it possible to pull from org/user locale?

Reply:
Or maybe it doesn't make a difference, because we're coping date data into another date field?